### PR TITLE
Add ckan machine class

### PR DESCRIPTION
**Pull request by boffbowsh**
Adds a new class of machine for running CKAN. This requires a persistent EBS volume due to the need to run Solr on this machine. The RAM requirement is high due to the number of harvester processes running.

Requires https://github.com/alphagov/govuk-aws-data/pull/150

### CODE REVIEW

File: terraform/projects/app-ckan/main.tf [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+  surgequeuelength_threshold     = "0"
+  healthyhostcount_threshold     = "0"
+}
+
+# Outputs
+# --------------------------------------------------------------
+
+output "ckan_elb_internal_address" {
+  value       = "${aws_elb.ckan_elb_internal.dns_name}"
+  description = "AWS' internal DNS name for the ckan ELB"
```
Comment (andrewgarner) [inline terraform/projects/app-ckan/main.tf]: Shall we say "CKAN" as it is an acronym "Comprehensive Knowledge Archive Network"?!

File: terraform/projects/app-ckan/remote_state.tf [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+variable "remote_state_infra_networking_key_stack" {
+  type        = "string"
+  description = "Override infra_networking remote state path"
+  default     = ""
+}
+
+variable "remote_state_infra_security_groups_key_stack" {
+  type        = "string"
+  description = "Override infra_security_groups stackname path to infra_vpc remote state "
```
Comment (andrewgarner) [inline terraform/projects/app-ckan/remote_state.tf]: Do these descriptions need trailing whitespace?

File: terraform/projects/app-ckan/main.tf (hunk 1 above)
Comment (boffbowsh) [reply terraform/projects/app-ckan/main.tf]: It's lowercase as it's the name of the ELB, which is lowercase.

Comment (andrewgarner) [reply terraform/projects/app-ckan/main.tf]: This is a description though?

Comment (boffbowsh) [reply terraform/projects/app-ckan/main.tf]: Yes but it's explicitly a description of which ELB it's pointing to, not what the ELB is used for.

Comment (andrewgarner) [reply terraform/projects/app-ckan/main.tf]: Fair enough 👍

File: terraform/projects/app-ckan/remote_state.tf (hunk 2 above)
Comment (boffbowsh) [reply terraform/projects/app-ckan/remote_state.tf]: Fixed

**Pull request merged**